tools/go-cli/go-whisk/whisk/activation.go

```diff
@@ -0,0 +1,186 @@
+        Debug(DbgError, "s.client.Do() error - HTTP req %s; error '%s'\n", req.URL.String(), err)
+        errStr := fmt.Sprintf("Request failure: %s", err)
+        werr := MakeWskErrorFromWskError(errors.New(errStr), err, EXITCODE_ERR_NETWORK, DISPLAY_MSG, NO_DISPLAY_USAGE)
+        return nil, resp, werr
+    }
+
+    return activations, resp, nil
+
+}
+
+func (s *ActivationService) Get(activationID string) (*Activation, *http.Response, error) {
+    // TODO :: for some reason /activations/:id only works with "_" as namespace
+    s.client.Namespace = "_"
+
+    route := fmt.Sprintf("activations/%s", activationID)
+
+    req, err := s.client.NewRequest("GET", route, nil)
+    if err != nil {
+        Debug(DbgError, "http.NewRequest(GET, %s) error: '%s'\n", route, err)
+        errStr := fmt.Sprintf("Unable to create HTTP request for GET '%s'; error: %s", route, err)
+        werr := MakeWskErrorFromWskError(errors.New(errStr), err, EXITCODE_ERR_GENERAL, DISPLAY_MSG, NO_DISPLAY_USAGE)
+        return nil, nil, werr
+    }
+
+    Debug(DbgInfo, "Sending HTTP request - URL '%s'; req %#v\n", req.URL.String(), req)
+
+    a := new(Activation)
+    resp, err := s.client.Do(req, &a)
+    if err != nil {
+        Debug(DbgError, "s.client.Do() error - HTTP req %s; error '%s'\n", req.URL.String(), err)
+        errStr := fmt.Sprintf("Request failure: %s", err)
+        werr := MakeWskErrorFromWskError(errors.New(errStr), err, EXITCODE_ERR_NETWORK, DISPLAY_MSG, NO_DISPLAY_USAGE)
+        return nil, resp, werr
+    }
+
+    return a, resp, nil
+
+}
+
+func (s *ActivationService) Logs(activationID string) (*Activation, *http.Response, error) {
+    // TODO :: for some reason /activations/:id/logs only works with "_" as namespace
+    s.client.Namespace = "_"
+    route := fmt.Sprintf("activations/%s/logs", activationID)
+
+    req, err := s.client.NewRequest("GET", route, nil)
+    if err != nil {
+        Debug(DbgError, "http.NewRequest(GET, %s) error: '%s'\n", route, err)
+        errStr := fmt.Sprintf("Unable to create HTTP request for GET '%s'; error: %s", route, err)
+        werr := MakeWskErrorFromWskError(errors.New(errStr), err, EXITCODE_ERR_GENERAL, DISPLAY_MSG, NO_DISPLAY_USAGE)
+        return nil, nil, werr
+    }
+
+    Debug(DbgInfo, "Sending HTTP request - URL '%s'; req %#v\n", req.URL.String(), req)
+
+    activation := new(Activation)
+    resp, err := s.client.Do(req, &activation)
+    if err != nil {
+        Debug(DbgError, "s.client.Do() error - HTTP req %s; error '%s'\n", req.URL.String(), err)
+        errStr := fmt.Sprintf("Request failure: %s", err)
+        werr := MakeWskErrorFromWskError(errors.New(errStr), err, EXITCODE_ERR_NETWORK, DISPLAY_MSG, NO_DISPLAY_USAGE)
+        return nil, resp, werr
+    }
+
+    return activation, resp, nil
+}
+
+func (s *ActivationService) Result(activationID string) (*Response, *http.Response, error) {
+    // TODO :: for some reason /activations only works with "_" as namespace
```

Comment:
All activations are stored in the default namespace. Using another namespaces will result in an empty response `[]`. The reason is that activating a trigger/rule/action cannot distinguish the context (namespace) within which to record the activation record.